src/librbd/mirror/snapshot/UnlinkPeerRequest.cc

```diff
@@ -63,13 +63,17 @@ void UnlinkPeerRequest<I>::unlink_peer() {
   m_image_ctx->image_lock.lock_shared();
   int r = -ENOENT;
   cls::rbd::MirrorSnapshotNamespace* mirror_ns = nullptr;
+  cls::rbd::SnapshotNamespace snap_namespace;
+  std::string snap_name;
   m_newer_mirror_snapshots = false;
   for (auto snap_it = m_image_ctx->snap_info.find(m_snap_id);
        snap_it != m_image_ctx->snap_info.end(); ++snap_it) {
     if (snap_it->first == m_snap_id) {
       r = 0;
       mirror_ns = std::get_if<cls::rbd::MirrorSnapshotNamespace>(
```

Comment:
It would be nicer to initialize `mirror_ns` directly from captured `snap_namespace` later:
```
  auto mirror_ns = std::get_if<cls::rbd::MirrorSnapshotNamespace>(
    &snap_namespace);
  if (mirror_ns == nullptr) {
    lderr(cct) << "not mirror snapshot (snap_id=" << m_snap_id << ")" << dendl;
```

Reply:
Sure that sounds better to me too, done.